lib/installer.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import os
+import subprocess
+
+
+class Installer(object):
+    """ Install all sorts of things. """
+
+    def __init__(self, handler_name, file_path):
+        self.handler_name = handler_name
+        self.file_path = file_path
+        self.requirements = self._parse_requirements()
+
+    def install(self):
+        install_handler = '_install_%s' % self.handler_name
+        getattr(self, install_handler)()
+
+    def remove(self):
+        remove_handler = '_remove_%s' % self.handler_name
+        getattr(self, remove_handler)()
+
+    def _install_apt(self):
+        self._run_command(
+            'apt-get update',
```

Comment:
I'd like to avoid running this twice if we have both apt and apt_build. Maybe something like `if not self.apt_updated: ...`?